src/openforms/formio/tests/validation/test_file.py

```diff
@@ -0,0 +1,45 @@
+from django.test import SimpleTestCase
+
+from ...typing import FileComponent
+from .helpers import extract_error, validate_formio_data
+
+
+class FileValidationTests(SimpleTestCase):
+    def test_file_max_files(self):
+        component: FileComponent = {
+            "type": "file",
+            "key": "foo",
+            "label": "Test",
+            "storage": "url",
+            "url": "http://localhost",
+            "useConfigFiletypes": False,
+            "filePattern": "",
+            "file": {"allowedTypesLabels": []},
+            "maxNumberOfFiles": 1,
+        }
+
+        is_valid, errors = validate_formio_data(
+            component,
+            {
+                "foo": [
+                    # truncated for brevity
+                    {
+                        "url": "some://url",
+                        "name": "my-foo-1.bin",
+                        "type": "application/foo",
+                        "originalName": "my-foo-1.bin",
+                    },
```

Comment:
probably best to define a factory for this that generates a representative data structure since this *will* get copy-pasted and cause problems in the future.